sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala

```diff
@@ -18,30 +18,32 @@
 package org.apache.spark.sql.catalyst
 
 /**
- * An identifier that optionally specifies a database.
+ * An identifier that optionally specifies a database and catalog.
  *
  * Format (unquoted): "name" or "db.name"
  * Format (quoted): "`name`" or "`db`.`name`"
  */
-sealed trait IdentifierWithDatabase {
+sealed trait IdentifierWithOptionalDatabaseAndCatalog {
   val identifier: String
 
   def database: Option[String]
 
+  def catalog: Option[String]
```

Comment:
Default to `None`?

Reply:
This is an abstract method definition and catalog is always implemented by a val.